src/TypeAssert.hack

```diff
@@ -41,18 +41,36 @@ function is_array_of_strings(mixed $value, string $field): array<string> {
   return $out;
 }
 
-function is_nullable_array_of_strings(
+function is_vec_like_of_strings(mixed $value, string $field): vec<string> {
+  invariant(
+    $value is vec<_> || \is_array($value),
```

Comment:
are we still getting non-vec arrays here?

Reply:
Yes we do, because of a legacy json decode in src/ConfigurationLoader.hack:31.

```
Fatal error: Uncaught exception 'HH\InvariantException' with message 'roots should be a vec<string>' in /path/to/hhvm-autoload/src/TypeAssert.hack:49
Stack trace:
#0 /path/to/hhvm-autoload/src/TypeAssert.hack(49): HH\invariant_violation()
#1 /path/to/hhvm-autoload/src/ConfigurationLoader.hack(55): Facebook\AutoloadMap\__Private\TypeAssert\is_vec_like_of_strings()
#2 /path/to/hhvm-autoload/src/ConfigurationLoader.hack(37): Facebook\AutoloadMap\ConfigurationLoader::fromData()
#3 /path/to/hhvm-autoload/src/ConfigurationLoader.hack(23): Facebook\AutoloadMap\ConfigurationLoader::fromJSON()
#4 /path/to/hhvm-autoload/src/builders/HHImporter.hack(57): Facebook\AutoloadMap\ConfigurationLoader::fromFile()
...
#10 (): Facebook\AutoloadMap\cli_main_UNSAFE()
#11 {main}
```

Reply:
This exception was generated by removing the `is_array()`. It works just fine with it.